### PR TITLE
[Improvement]: disable introspection by default for production environment

### LINKED ISSUE #473: [Improvement]: disable introspection by default for production environment
### Improvement description

By default introspection for GraphQL is enabled, this applies to all environments. Because of the ease of setting up a GraphQL endpoint human errors can occur by accidentally adding sensitive data.

When this happens in production environments it is very easy to introspect the entire GraphQL endpoint and find out about this kind of data.

I think it would be a great improvement when introspection is disabled by default on production environments. Introspection could be enabled when you're logged into the Pimcore admin on a production environment and when you have the right permissions.

### PR DESCRIPTION
**Pull request by mcop1**
resolves #473

### CODE REVIEW

File: src/Migrations/Version20220322144548.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+use Pimcore\Bundle\DataHubBundle\Configuration;
+
+final class Version20211108160248 extends AbstractMigration
+{
+    private function migrateConfiguration(bool $up)
+    {
+        $configs = Configuration::getList();
```
Comment (fashxp) [inline src/Migrations/Version20220322144548.php]: this will deliver all configurations, not only graphql ones ... we need to filter for type graphql

**Review by fashxp — CHANGES_REQUESTED** [review]

File: src/Migrations/Version20220322144548.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\DataHubBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+use Pimcore\Bundle\DataHubBundle\Configuration;
+
+final class Version20211108160248 extends AbstractMigration
```
Comment (fashxp) [inline src/Migrations/Version20220322144548.php]: - migration actually needs to be located in  Pimcore\Bundle\DataHubBundle\Migrations\PimcoreX namespace
- filename and classname need to match
- should extend from `BundleAwareMigration`  ... to make sure it is not executed when bundle is not installed

**Review by fashxp — CHANGES_REQUESTED** [review]

**Review by fashxp — CHANGES_REQUESTED** [review]
migration cannot update the configs, if they are yaml files located in custom locations like bundles etc. 

I guess we need to get rid of the migration and introduce a BC layer ... if config value not set, set it to true.

File: src/Migrations/Version20220322144548.php (hunk 2 above)
Comment (mcop1) [reply src/Migrations/Version20220322144548.php]: > * migration actually needs to be located in  Pimcore\Bundle\DataHubBundle\Migrations\PimcoreX namespace
> * filename and classname need to match
> * should extend from `BundleAwareMigration`  ... to make sure it is not executed when bundle is not installed

As discussed, I changed the checkbox from "Enable Introspection" to "Disable Introspection". Default: Introspection is enabled.

**Comment by mcop1** [comment]
> migration cannot update the configs, if they are yaml files located in custom locations like bundles etc.
> 
> I guess we need to get rid of the migration and introduce a BC layer ... if config value not set, set it to true.

As discussed, I changed the checkbox from "Enable Introspection" to "Disable Introspection". Default: Introspection is enabled. No more migration needed.

**Review by fashxp — APPROVED** [review]

**Pull request merged**